### PR TITLE
[android][audio] WIP expo-audio for android

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-12535

# How
Implement the initial work on android for expo-audio

# Test Plan
bare-expo + NCL

### CODE REVIEW

File: packages/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt [hunk 1]
```diff
@@ -1,10 +1,500 @@
...
+    }
+
+    OnDestroy {
+      val uiManager = appContext.legacyModule<UIManager>()
+      uiManager?.unregisterLifecycleEventListener(this@AudioModule)
+    }
+
+    Class(AudioPlayer::class) {
+      Constructor { source: AudioSource? ->
+        val builder = OkHttpClient().newBuilder()
```
Comment (lukmccall) [inline packages/expo-audio/android/src/main/java/expo/modules/audio/AudioModule.kt:111]: Clients should be shared (check https://square.github.io/okhttp/5.x/okhttp/okhttp3/-ok-http-client/index.html). Creating a new one for each session isn't a good idea. There should be a construction that allows you to customize things like headers.

File: packages/expo-audio/android/src/main/java/expo/modules/audio/AuidoPlayer.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      Player.STATE_ENDED -> "ended"
+      else -> "unknown"
+    }
+  }
+
+  private fun sendEventOnJSThread(vararg args: Any?) {
+    appContext?.executeOnJavaScriptThread {
+      sendEvent("onPlaybackStatusUpdate", *args)
+    }
+  }
```
Comment (lukmccall) [inline packages/expo-audio/android/src/main/java/expo/modules/audio/AuidoPlayer.kt:116]: Wrapping sendEvent with `executeOnJavaScriptThread` is not longer needed.

File: packages/expo-audio/android/build.gradle [hunk 3]
```diff
@@ -16,4 +16,19 @@ android {
...
+  mavenCentral()
 }
+
+dependencies {
+  implementation("androidx.media3:media3-exoplayer:1.3.1")
+  implementation("androidx.media3:media3-exoplayer-dash:1.3.1")
+  implementation("androidx.media3:media3-datasource-okhttp:1.3.1")
+  implementation("androidx.media3:media3-exoplayer-hls:1.3.1")
+  implementation("androidx.media3:media3-exoplayer-smoothstreaming:1.3.1")
+}
```
[the remark this answers is not in the document]
Comment (alanjhughes) [reply packages/expo-audio/android/build.gradle:34]: Does spotless only run on Kotlin files? I thought it did these files too but maybe not?

File: packages/expo-audio/src/utils/options.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    newOptions = {
+      ...newOptions,
+      ...options.ios,
+    };
+  } else if (Platform.OS === 'android') {
+    newOptions = {
+      ...newOptions,
+      ...options.android,
+    };
+  }
```
Comment (alanjhughes) [inline packages/expo-audio/src/utils/options.ts:23]: Finally else will be added when adding web support

**Review by lukmccall — APPROVED** [review]

**Pull request merged**